src/lib.rs

```diff
@@ -112,28 +108,28 @@ impl Firebase {
     /// // A new reference to /friends/yasha/messages
     /// let messages = yasha.at("messages").unwrap();
     pub fn at(&self, add_path: &str) -> Result<Self, ParseError> {
-        let mut url = (*self.url).clone();
+        let mut url = self.url.clone();
 
         { // Add path to original path, already checked for path.
-            let mut path = url.path_mut().unwrap();
             // Remove .json from the old path's end.
-            if let Some(end) = path.pop() {
-                path.push(end.trim_right_matches(".json").to_string());
-            }
+            let last = url.path_segments().expect("last segment").last().unwrap().to_string();
+            let mut path = url.path_segments_mut().expect("path segments");
```

Comment:
Yes, it is guaranteed to have a path segment (see http://servo.github.io/rust-url/url/struct.Url.html#method.path_segments). Though, ff the Url is a "cannot-be-a-base URLs" (whatever that means) then this fails.